### PR TITLE
couple small features

### PR DESCRIPTION
**Pull request by hugohabicht01**
- make it possible to clear the screen and preserve command history
- use a cleaner solution using `nextTick` for screen clearing, opposed to `setTimeout`
- have basic tab autocomplete
- make the terminal scrollable instead of it expanding in size, with autoscroll to the last command output

### CODE REVIEW

File: components/Terminal/commands/registry.ts [hunk 1]
```diff
@@ -20,3 +20,7 @@ export function get_command(name: string) {
 export function get_help(name: string) {
   return registry[name]?.help
 }
+
+export function get_all_commands() {
```
Comment (moddedTechnic) [inline components/Terminal/commands/registry.ts:24]: Do we want camelCase names or snake_case as at the moment its not consistent

File: components/Terminal/index.vue [hunk 2]
```diff
@@ -40,9 +44,25 @@ function handleCommand(command: string): string | undefined {
 function handleInput(event: KeyboardEvent) {
   const { key } = event;
 
-  if (key === "Tab") return;
   event.preventDefault();
 
+  if (key === "Tab") {
```
Comment (moddedTechnic) [inline components/Terminal/index.vue:49]: N.b. Tab is used by default in browsers for keyboard navigation, perhaps its worth providing an escape method? (Shift+Tab or something)

Comment (hugohabicht01) [reply components/Terminal/index.vue:49]: how about esc, which we could use to set focus to the next element after the terminal. if we set in front of it everything after it becomes inaccessible

File: components/Terminal/commands/registry.ts (hunk 1 above)
Comment (hugohabicht01) [reply components/Terminal/commands/registry.ts]: döñé

**Review by MaxAFriedrich — APPROVED** [review]

File: components/Terminal/index.vue (hunk 2 above)
Comment (moddedTechnic) [reply components/Terminal/index.vue:49]: That would work.  When we do the initial message, we probably want to alert the user to use escape to exit, but that's not difficult

**Review by moddedTechnic — APPROVED** [review]

File: components/Terminal/index.vue (hunk 2 above)
Comment (hugohabicht01) [reply components/Terminal/index.vue:49]: when someone is pressing tab or when someone is pressing esc?

Comment (moddedTechnic) [reply components/Terminal/index.vue:49]: I was thinking of doing some kind of welcome message when the terminal starts. Something like this (but probably shorter):
20230823_12h13m13s_grim

**Pull request merged**